app/controllers/users/backup_code_setup_controller.rb

```diff
@@ -0,0 +1,52 @@
+module Users
+  class BackupCodeSetupController < ApplicationController
+    before_action :authenticate_user!
+    before_action :confirm_two_factor_authenticated, if: :two_factor_enabled?
+    before_action :ensure_backup_codes_in_session, only: %i[create download]
+
+    def index
+      generate_codes
+      result = BackupCodeSetupForm.new(current_user).submit
+      analytics.track_event(Analytics::BACKUP_CODE_SETUP_VISIT, result.to_h)
+    end
+
+    def create
+      analytics.track_event(Analytics::BACKUP_CODE_CREATED)
+      mark_user_as_fully_authenticated
```

Comment:
Personally, I'd wait until after we know the backup codes are saved, but I don't think this is a blocker.